email_notification/sender/email_notification_builder.py

```diff
@@ -210,31 +213,25 @@ def _init_context_attr_map(self):
                 raise EmailBuilderConfigError(
                     "Email context variable %s did not had _get method defined." % key
                 )
-            self.context_attr_map[key] = value
+            self.context_attr_map[key] = value()
 
     def validate_template(self):
         validator = self.validator()
-        validator.validate_string(self.template.subject, self.context_attr_map)
-        validator.validate_string(self.template.content, self.context_attr_map)
 
         html_content = self._get_html_content(self.template)
         if html_content:
-            validator.validate_string(html_content)
+            validator.validate_string(html_content, self.context_attr_map, env=self.env)
 
-    def get_context(self):
-        context_dict = {}
-        for key, value in self.context_attr_map.items():
-            if callable(value):
-                context_dict[key] = value()
-                continue
-
-            context_dict[key] = getattr(self.reservation, value, None)
-
-        return Context(context_dict)
+        validator.validate_string(
+            self.template.subject, self.context_attr_map, env=self.env
+        )
+        validator.validate_string(
```

Comment:
`validator.validate_string` is called twice here.

Reply:
good catch, poorly rebased, I'll clean this

Reply:
oh, I backup a bit, there is a validation check for subject and content (the html content is its own content) so it is supposed be like this.

Reply:
Funny, I was thinking that these are NOT the same line. To be sure to not look like a blind idiot, I read them like 100 times. I was sure they are identical.

I just learned I'm a blind idiot 🧑‍🦯🤪